GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -351,11 +351,11 @@ async Task UpdateCommitMessageAsync()
             {
                 CommitData data = _commitDataManager.CreateFromRevision(_revision, _children);
 
-                if (_revision.Body is null || (AppSettings.ShowGitNotes && !_revision.HasNotes))
+                if (_revision.Body is null || ((AppSettings.ShowGitNotesColumn || AppSettings.ShowGitNotes) && _revision.Notes is null))
                 {
                     _commitDataManager.UpdateBody(data, appendNotesOnly: _revision.Body is not null, out _);
                     _revision.Body = data.Body;
-                    _revision.HasNotes = true;
+                    _revision.Notes = data.Notes;
```

Comment:
```suggestion
                    _revision.Notes = data.Notes ?? "";
```
in order to indicate that we know that there are no notes.